packages/plugins/on-resolve/src/index.ts

```diff
@@ -32,14 +32,18 @@ export type OnResolve<PluginContext extends Record<string, any> = {}> = (
   options: OnResolveOptions<PluginContext>,
 ) => PromiseOrValue<AfterResolver | void>;
 
+export type UseOnResolveOptions = {
+  introspection: Boolean;
```

Comment:
Just "introspection" doesn't say much about what this property does. I'd prefer to name it `skipIntrospection` (and document the default) to make it more clear.

Also, avoid using the `Boolean` alias type (capital B).

```suggestion
  /**
  * Skip executing the `onResolve` hook on introspection queries.
  *
  * @default true
  */
  skipIntrospection: boolean;
```